### PR TITLE
feat(rpc): add failed swaps to PlaceOrderSync

### PR DESCRIPTION
**Pull request by sangaman**
This adds a list of failed swaps to the `PlaceOrderSync` response to match the behavior of the asynchronous `PlaceOrder` call which returns information about failed swaps.

Solves part of #807.

### CODE REVIEW

File: lib/orderbook/types.ts [hunk 1]
```diff
@@ -15,6 +15,7 @@ export type PlaceOrderResult = {
   internalMatches: OwnOrder[];
   swapSuccesses: SwapSuccess[];
   remainingOrder?: OwnOrder;
+  swapFailures: PeerOrder[];
```
Comment (moshababo) [inline lib/orderbook/types.ts]: Minor: should jump this one line up, next to `swapSuccesses`. also in other sections in this PR.

File: lib/orderbook/OrderBook.ts [hunk 2]
```diff
@@ -371,6 +375,7 @@ class OrderBook extends EventEmitter {
           onUpdate && onUpdate({ type: PlaceOrderEventType.SwapSuccess, payload: swapSuccess });
         } catch (err) {
           this.logger.warn(`swap for ${portion.quantity} failed during order matching, will repeat matching routine for failed swap quantity`);
+          swapFailures.push(maker);
```
Comment (moshababo) [inline lib/orderbook/OrderBook.ts]: We should add the swap reason here (and in `onUpdate` callback as well), and not to return merely the taker order. Did you plan to keep this for a different PR?

Comment (kilrau) [reply lib/orderbook/OrderBook.ts]: Same question, see our discussion on discord.

File: lib/orderbook/types.ts (hunk 1 above)
Comment (sangaman) [reply lib/orderbook/types.ts]: OK I was actually thinking about doing the same, so I'll go ahead with that since you had the same thought.

File: lib/orderbook/OrderBook.ts (hunk 2 above)
Comment (sangaman) [reply lib/orderbook/OrderBook.ts]: Yes I was, but I can add it to this PR in a separate commit so it completely closes #807.

Comment (sangaman) [reply lib/orderbook/OrderBook.ts]: Actually I'll open a separate PR because it's not a very simple change and I think it'd be better to merge this then discuss more changes.
